src/domdiv/config_options.py

```diff
@@ -908,79 +927,55 @@ def clean_opts(options):
-        options.expansions = [
-            item.lower() for sublist in options.expansions for item in sublist
-        ]
+        options.expansions = flatten_lower_and_deduplicate(options.expansions)
     if "none" in options.expansions:
         # keyword to indicate no options.  Same as --expansions without any expansions given.
-        options.expansions = []
+        options.expansions.clear()
 
     if options.exclude_expansions:
-        # options.exclude_expansions is a list of lists.  Reduce to single lowercase list
-        options.exclude_expansions = [
-            item.lower() for sublist in options.exclude_expansions for item in sublist
-        ]
+        # options.exclude_expansions is a list of lists.  Reduce to single lowercase set
+        options.exclude_expansions = flatten_lower_and_deduplicate(
+            options.exclude_expansions
+        )
 
     if options.edition is None:
-        # set the default
+        # This is the default unless there were also no expansions given.
         options.edition = "all"
 
-    if options.fan is None:
-        # No instance given, so default to no Fan expansions
-        options.fan = []
-    else:
-        # options.fan is a list of lists.  Reduce to single lowercase list
-        options.fan = [item.lower() for sublist in options.fan for item in sublist]
+    options.fan = flatten_lower_and_deduplicate(options.fan)
     if "none" in options.fan:
         # keyword to indicate no options.  Same as --fan without any expansions given
-        options.fan = []
+        options.fan.clear()
 
-    if options.only_type_any is None:
-        # No instance given, so default to empty list
-        options.only_type_any = []
-    else:
-        # options.only_type_any is a list of lists.  Reduce to single lowercase list
-        options.only_type_any = list(
-            set([item.lower() for sublist in options.only_type_any for item in sublist])
-        )
+    options.only_type_any = flatten_lower_and_deduplicate(options.only_type_any)
+    options.only_type_all = flatten_lower_and_deduplicate(options.only_type_all)
 
-    if options.only_type_all is None:
-        # No instance given, so default to empty list
-        options.only_type_all = []
-    else:
-        # options.only_type_any is a list of lists.  Reduce to single lowercase list
-        options.only_type_all = list(
-            set([item.lower() for sublist in options.only_type_all for item in sublist])
-        )
-
-    if options.group_global is None:
-        options.group_global = []
-    elif not any(options.group_global):
-        # option given with nothing indicates all possible global groupings
-        options.group_global = db.get_global_groups()[1]
-    else:
-        # options.group_global is a list of lists.  Reduce to single lowercase list
-        options.group_global = [
-            item.lower() for sublist in options.group_global for item in sublist
-        ]
+    legacy_groups = set()
```

Comment:
Not in this PR, but we should deprecate these. There are more than enough options already.

Reply:
Amen. I just don't know what's being used by the website, so it's hard for me to know what can and can't be removed.

Reply:
You can see it here: https://github.com/sumpfork/bgtools_web/blob/main/assets/lambda/domdiv_form.py#L30 - it's pretty custom, and I'll likely have to update that anyway when I deploy all these changes. I had been working to fully automate representing all relevant options of the script on the site directly (as an 'advanced' tab), but it ends up being somewhat cumbersome, so this is what it is for now.